core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -1082,29 +1083,38 @@ private void syncLocalToRemote() {
                 //    version with an update containing the replacement document with the version
                 //    counter incremented by 1.
                 final BsonDocument translatedUpdate = new BsonDocument();
+                final BsonDocument sets = new BsonDocument();
+                final BsonDocument unsets = new BsonDocument();
                 if (!localChangeEvent.getUpdateDescription().getUpdatedFields().isEmpty()) {
-                  final BsonDocument sets = new BsonDocument();
                   for (final Map.Entry<String, BsonValue> fieldValue :
                       localChangeEvent.getUpdateDescription().getUpdatedFields().entrySet()) {
                     sets.put(fieldValue.getKey(), fieldValue.getValue());
                   }
-                  sets.put(DOCUMENT_VERSION_FIELD, nextVersion);
-                  translatedUpdate.put("$set", sets);
                 }
                 if (!localChangeEvent.getUpdateDescription().getRemovedFields().isEmpty()) {
-                  final BsonDocument unsets = new BsonDocument();
                   for (final String field :
                       localChangeEvent.getUpdateDescription().getRemovedFields()) {
                     unsets.put(field, BsonBoolean.TRUE);
                   }
                   translatedUpdate.put("$unset", unsets);
                 }
+                if (!sets.isEmpty() || !unsets.isEmpty()) {
+                  // set the document version if any changes were made by this update
+                  sets.put(DOCUMENT_VERSION_FIELD, nextVersion);
+                  translatedUpdate.put("$set", sets);
+                }
 
                 final RemoteUpdateResult result;
                 try {
+                  if (translatedUpdate.isEmpty()) {
+                    // if the translated update is empty, then this update is a noop, and we
```

Comment:
[q] Should we log here? Also, maybe we should just check if it's a noop earlier on so that getting this branch is technically impossible.

Reply:
added the check earlier in the `UPDATE` and added a log statement, but this branch was definitely possible to begin with. Any update that matches documents but doesn't end up changing them locally will produce a noop event until we modify the update functions to not emit events for noop updates (and also properly return a modifiedCount that reflects these noop updates).